Fix Photo G-code CHDK_PIN trigger time

RaulLazaro opened the pull request:
### Requirements
Enabling the Photo G-code in Configuration_adv.h using CHDK_PIN and with PHOTO_DELAY_MS and PHOTO_SWITCH_MS defined.

### Description
I realized when using the G-code M240 that the parameter [D <ms>] did not work and that the CHDK_PIN high was maintained until the end of the parameter [P <ms>] and start the movement.

### Benefits
Now this is solved and work as expected, I tested in a SKR 1.3 using pin P1_26.

RaulLazaro committed:
Solution of trigger time CHDK_PIN for Photo G-code

Enabling the Photo G-code in Configuration_adv.h using CHDK_PIN and with PHOTO_DELAY_MS and PHOTO_SWITCH_MS defined, I realized when using the G-code M240 that the parameter [D <ms>] did not work and that the CHDK_PIN high was maintained until the end of the parameter [P <ms>] and start the movement.

This is solved and has been tested in a SKR 1.3 using pin P1_26.

thinkyhead commented:
This changes `M240` into a blocking G-code. The intent was to allow for a shutter to be held for a short or even a very long period without interrupting motion.

thinkyhead committed:
Fix M240

RaulLazaro commented:
With this you are not taking into account the parameters sent by g-code, for example if you have defined PHOTO_SWITCH_MS to 500 and you send `M240 D1000` it will continue doing 500ms, and now that will also happen with the parameter P.

And if the problem is changing `M240` into a blocking G-code by using safe_delay, this is still used when PHOTO_SWITCH_POSITION is activated right in the previous lines.

RaulLazaro committed:
Fix M240

Added `#include "../../../module/planner.h"` to be able to use `idle ()`.
Use of `parser.intval()` to take into account the parameters sent.

Compiled correctly and tested, now I understand how it wanted to work, you can keep CHDK_PIN high while moving.moving.

thinkyhead commented:
Thanks for bringing in the new parameters. Looks pretty complete now!

thinkyhead committed:
Should be ok w/out extra planner.h include

thinkyhead commented:
> this is still used when `PHOTO_SWITCH_POSITION` is activated right in the previous lines.

The `CHDK_PIN` is used only when you want to a trigger a camera directly, and it has a specific duration requirement for triggering the camera shutter that is attached to the pin to get a single successful shutter-click.

The `PHOTO_SWITCH_POSITION` is used when you have an endstop switch connected directly to a camera, when there is no pin associated with `M240` at all. In that case, it is necessary for `M240` to do certain moves and stops to have the carriage trigger the switch, and these moves and pauses are of course done in-place, and are supposed to be blocking.

The `safe_delay` before the trigger of `CHDK_PIN` would be fine in the case of combining it with `PHOTO_SWITCH_POSITION` — though this combination would not be typical.

RaulLazaro reviewed (commented):
Without planner.h I get an error compiling:

> Marlin/src/gcode/feature/camera/M240.cpp: In static member function 'static void GcodeSuite::M240()':
Marlin/src/gcode/feature/camera/M240.cpp:164:42: error: 'idle' was not declared in this scope
       while (PENDING(millis(), timeout)) idle();
       while (PENDING(millis(), timeout)) idle();
                                          ^~~~
Marlin/src/gcode/feature/camera/M240.cpp:164:42: note: suggested alternative: 'id_t'
       while (PENDING(millis(), timeout)) idle();
                                          ^~~~
                                          id_t
*** [.pio/build/LPC1768/src/src/gcode/feature/camera/M240.cpp.o] Error 1

thinkyhead commented:
I see! For `idle()` the correct include would be `../../../Marlin.h`.
